test_util/src/envfile.rs

```diff
@@ -0,0 +1,156 @@
+use super::*;
+
+/// Create the default envfile contents.
+///
+/// [`DEFAULT_TEST_KEY`] set as [`DEFAULT_TEST_VALUE`]
+///
+/// [`DEFAULT_EXISTING_KEY`] set as [`DEFAULT_OVERRIDING_VALUE`]
+pub fn create_default_envfile() -> String {
+    format!(
+        "{DEFAULT_TEST_KEY}={DEFAULT_TEST_VALUE}\n{DEFAULT_EXISTING_KEY}={DEFAULT_OVERRIDING_VALUE}",
+    )
+}
+
+/// Invalid due to missing `=` between key and value.
+pub fn create_invalid_envfile() -> String {
+    format!(
+        "{DEFAULT_TEST_KEY}{DEFAULT_TEST_VALUE}\n{DEFAULT_EXISTING_KEY}{DEFAULT_OVERRIDING_VALUE}",
+    )
+}
+
+/// Create an envfile with custom key-value pairs.
+///
+/// ## Example
+///
+/// ```no_run
+/// # use dotenvy_test_util::create_custom_envfile;
+/// let contents = create_custom_envfile(&[
+///     ("CUSTOM_KEY", "test_val"),
+///     ("ANOTHER_KEY", "another_val"),
+/// ]);
+/// assert_eq!(contents, "CUSTOM_KEY=test_val\nANOTHER_KEY=another_val\n");
+/// ```
+pub fn create_custom_envfile(env_vars: &[(&str, &str)]) -> String {
+    let mut efb = EnvFileBuilder::new();
+    efb.add_vars(env_vars);
+    efb.into_owned_string()
+}
+
+/// Advanced test-envfile constructor.
+///
+/// Represented as bytes to allow for advanced manipulation and BOM testing.
+#[derive(Debug, Default)]
+pub struct EnvFileBuilder {
```

Comment:
I am good with either `AdvancedEnvFile` or `AdvEnvFile`. I personally prefer shorter names; they are more adventurous (`AdventurousEnvFile`?)

Reply:
`Efb` is such a good acronym though. `EnvFileBuilder` has grown on me!

Reply:
Despite `efb` working really well, I think `AdvEnvFile` works best as a full name and will action it. It is only tangentially related to the "builder pattern," the use is far more in line with that of a byte buffer/vector.

Reply:
I am currently simplifying this crate. Removing the "default" code I have come to realise this should just be `EnvFile` and be the primary way one is constructed.

A `String`, `&str`,  `Vec<u8>`, `&[u8]` or `EnvFileBuilder`, can already be passed to `TestEnv::add_env_file`. So why not just call this `EnvFile` as it will be the only way of constructing one besides just using the primitive types.